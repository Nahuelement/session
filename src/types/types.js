

export const types = {
    login:'[Auth] Login',
    logout:'[Auth] Logout',

    uisetError: '[UI] set Error',
    uiRemoveError: '[UI] Remove Error',

    uiStartLoading: '[UI] Start loading',
    uiFinishLoading: '[UI] Finish loading',


    notesAddNew : '[Notes] New note',
    notesActive : '[Notes] set active note',
    notesLoad : '[Notes] Load note',
    notesUpdated : '[Notes] Updated note',
    notesFileUrl : '[Notes] updated image url',
    notesDelete : '[Notes] Delete note',
    notesLogoutCleaning : '[Notes] logout Cleaning',
    activeModal : '[UI] activar modal',
    endModal : '[UI] desactivar modal'
}
