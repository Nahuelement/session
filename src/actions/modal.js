import { types } from "../types/types"




export const openModal =() =>({
    type : types.activeModal,
    
})
export const endModal =() =>({
    type : types.endModal,
    
})