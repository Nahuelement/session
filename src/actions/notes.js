import { db } from "../firebase/fireConfig";
import { types } from "../types/types";
import { collection ,addDoc, doc, setDoc, deleteDoc} from "firebase/firestore";
import { loadNotes } from "../helpers/loadNotes";
import Swal from "sweetalert2";
import { fileUpload } from "../helpers/fileUpload";

// import cloudinary from 'cloudinary'

// // react-journal

//  cloudinary.config({ 
//     cloud_name: 'nahuelement', 
//     api_key: '723161997253356', 
//     api_secret: 'c2WpIaroeOWZNhdCdN8S0CQp-WE',
//     secure: true
//  });

export const startNewNotes = () =>{

    return async(dispatch, getState) =>{ // getState para obtener el estado

        const {uid} = getState().auth;
        

        const newNote = {
            title: '', 
            body: '',
            date: new Date().getTime(),
            url: ''
        }
        try{
            const docRef = await addDoc(collection(db,`${uid}/journal/notes`),newNote)
            dispatch(activeNote(docRef.id,newNote))
            // dispatch(startLodingNotes(uid)) IDEA MIA 
            dispatch(addNewNote(docRef.id,newNote))

        }
        catch (err) {
            console.log(err)
        } 
    }
}

export const activeNote = (id, note) => ({

    type:types.notesActive,
    payload:{
        id,
        ...note
    }
})

export const addNewNote = (id,note) => ({

    type:types.notesAddNew,
    payload:{
        id,
        ...note
    }
})


export const startLodingNotes = (uid) => {

    return async(dispatch) =>{

        try{
            const notes = await loadNotes(uid)
            dispatch(setNotes(notes))
        }
        catch(err) {
            console.log(err)

        }        
    }
}



export const setNotes = (notes) => ({

    type:types.notesLoad,
    payload:notes
    
}) 

export const startSaveNote = (note) => { 

    return async (dispatch, getState) =>{

        const {uid} = getState().auth; 
        
         if(!note.url){
             delete note.url;
         } 

        const noteToFirestore = {...note};
        delete noteToFirestore.id
        // console.log(noteToFirestore)
        
        try{
            await setDoc(doc(db,`${uid}/journal/notes/${note.id}`),noteToFirestore)
            Swal.fire('Guardado!',note.title,'success')
            dispatch(refreshNote(note.id,{...note}))
        }
        catch(err) {
            console.log(err)
        }
        
        


        


       
            
        
            
       
       
    }   

}
export const refreshNote = (id,note) =>({
    type:types.notesUpdated,
    payload:{
        id,
        note
    }

})

export const startUpLoading = (file) =>{
    return async(dispatch, getState) =>{

        const {active: activeNote} = getState().notes;
        
        Swal.fire({ 
            title: 'Uploading...',
            text:' Plase wait...',
            allowOutsideClick:false,
            allowEnterKey: () =>{
                Swal.showLoading();
            }
        }) 

        let fileUrl  = await fileUpload(file)
        
        Swal.close()
         if(!fileUrl){
         fileUrl= 'www.koi.partedelaprueba.porproblemasconJest().com'
        }
        activeNote.url = fileUrl
        console.log(activeNote)
       

        dispatch(startSaveNote(activeNote))
        
    }
}

export const startDeleting = (id) => {
    return async(dispatch, getState) =>{

        const uid = getState().auth.uid

        try{
            await deleteDoc(doc(db,`${uid}/journal/notes/${id}`)) 
            dispatch(deleteNotes(id))
        }
        catch(err) {
            console.log(err)
            
        }
        

        
    }
}

export const deleteNotes = (id) => ({
    type:types.notesDelete,
    payload:id
})

export const noteLogout = () => ({
    type:types.notesLogoutCleaning,
})