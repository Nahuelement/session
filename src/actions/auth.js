import { googleAuthProvider,facebookAuthProvider} from "../firebase/fireConfig"
import { getAuth, signInWithPopup, GoogleAuthProvider ,signInWithEmailAndPassword,
         FacebookAuthProvider,createUserWithEmailAndPassword,updateProfile,
         signOut} from "firebase/auth";
import { types } from "../types/types"
import { uiFinishLoading, uiStartLoading } from "./ui";
import Swal from "sweetalert2";
import { noteLogout } from "./notes";




export const startLogin = (email, password) => {

    return(dispatch) =>{
        dispatch(uiStartLoading())


        signInWithEmailAndPassword(getAuth(), email, password)
        .then(({user}) =>{
            dispatch(login(user.uid,user.displayName));
            dispatch(uiFinishLoading())
       
        }).catch(e => {
            
            // const errorCode = e.code;
            const errorMessage = e.message;

            let message = errorMessage;

            if(String(errorMessage)==='Firebase: Error (auth/invalid-email).'){
                 message = 'ingrese email valido'

            }else if (String(errorMessage)==='Firebase: Error (auth/wrong-password).'){
                 message = 'Password incorrecto'
                 
            }else if (String(errorMessage)==='Firebase: Error (auth/internal-error).'){
                 message = 'Ingrese email y contraseÃ±a'

                 
            }else if (String(errorMessage)==='Firebase: Error (auth/user-not-found).'){
                message = 'Email no registrado'
                
           }
           
            dispatch(uiFinishLoading())

            Swal.fire('Error',message,'error')
            
      })

    
}
}

export const startRegisterWhitEmailPassword  = (email, password,name) => {

    return(dispatch) =>{

    createUserWithEmailAndPassword(getAuth(),email, password)
    .then(async({user}) =>{
        
        await updateProfile(getAuth().currentUser,{displayName:name})
        
        dispatch(login(user.uid,user.displayName))
       
        
    })
    .catch((error) => {
        // const errorCode = error.code;
        const errorMessage = error.message;

        let message;
        if(String(errorMessage)==='Firebase: Error (auth/email-already-in-use).'){
            message = 'Usuario ya registrado con este email'

        }
       
        Swal.fire('Error',message,'error')

    }
    )
}}

export const starGoogleLogin = () => {
    
    return(dispatch) => {
        
       
        signInWithPopup(getAuth(),googleAuthProvider)
        .then(({user}) =>{
            dispatch(login(user.uid,user.displayName))
            
        }).catch((e) =>{
            
            GoogleAuthProvider.credentialFromError(e);
            
        })
        
    
    }}

    export const starFacebookLogin = () => {
    
        return(dispatch) => {
           
            signInWithPopup(getAuth(),facebookAuthProvider)
            .then((result) =>{
                const user = result.user;

                console.log(user)
            }).catch(FacebookAuthProvider.credentialFromError)
        }}
            

  

export const login = (uid, displayName) => ({

   
        type: types.login,
        payload: {
            uid,
            displayName
        }
})
export const startLogout = () => {

   return async(dispatch)=> {

    await signOut(getAuth())
    dispatch(logout())
    dispatch(noteLogout())

   }
    
}

export const logout = () => ({

    type: types.logout,
})
