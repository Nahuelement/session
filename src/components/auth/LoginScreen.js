import React from 'react'
import { useDispatch, useSelector } from 'react-redux'

import {  starGoogleLogin, startLogin } from '../../actions/auth'
import { useForm } from '../../hooks/useForm'



export const LoginScreen = () => {

    const dispatch = useDispatch()//PUEDE HACER DISPATCH DESDE CUALQUIER LUGAR DE LA APLICACION

    const {loading} = useSelector(state =>state.ui)
    const [formValues, HandleInputChange] = useForm({
        email: '',
        password: ''
    });

    const {email , password} = formValues;

    const handleLogin = () => {
        
        
        dispatch(startLogin(email,password))
      

    }
    const handleLoginGoogle = (e) => {
        
        dispatch(starGoogleLogin())

    }
    



    return (
        <>
        
            <h3 className="auth__title ">Login Mock</h3>
            <form onSubmit={handleLogin}>

                <input 
                type="text"
                placeholder="mail"
                name="email"
                className="auth__input"
                autoComplete = 'off'
                value = {email}
                onChange={HandleInputChange}
                />
                <input 
                type="password"
                placeholder="password"
                name="password"
                className="auth__input"
                autoComplete = 'off'
                value = {password}
                onChange={HandleInputChange}
                />


                <button
                type="submit"
                className="btn btn-primary btn-block"
                onClick={handleLogin}
                
                disabled = {loading}
                
                
                >Login
                
                </button>

      
                <div className="auth__social-networks">
                    <p>
                        Login con Google
                    </p>
                    <div 
                        className="google-btn"
                        onClick={handleLoginGoogle}
                        disabled = {loading}
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                    <br/>
                    
                </div>

                {/* <Link 
                to="/auth/register"
                className="link"
                >
                    Crear nueva cuenta
                </Link> */}


            </form>
            
        </>
    )
}
