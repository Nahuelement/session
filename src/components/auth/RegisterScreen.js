import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'
import validator from 'validator'

import { useDispatch, useSelector } from 'react-redux'
import { removeError, setError } from '../../actions/ui'
import { startRegisterWhitEmailPassword } from '../../actions/auth'
import Swal from "sweetalert2";



export const RegisterScreen = () => {



    const dispatch = useDispatch()
    const {msgError} = useSelector(state =>state.ui)

    
    // useEffect(() => {

    //     if (msgError){
    //     Swal.fire('Error',String(msgError),'error')}
        
        
    // }, [dispatch])


    const [ formvalues, handleInputChange ] = useForm({
        name:'esto es ',
        email:'un@ejemplo',
        password:'de ',
        confirm:'de un register'
    })
    const {name, email, password,confirm} = formvalues



    const handlenRegister = (e) => {
        e.preventDefault()
        

        if(isFormValid()){
            dispatch(startRegisterWhitEmailPassword(email, password,name))
            
        }
            
        
        

        
        
    }

    const isFormValid = () => {
        if(name.trim().length ===0){
            dispatch(setError('Nombre de usuario requerido'))
            
            Swal.fire('Error','Nombre de usuario requerido','error')
            
            return false
        }else if (!validator.isEmail(email)){
            dispatch(setError('Email valido requerido'))
            Swal.fire('Error','Email valido requerido','error')
            
            return false

        }else if (password!==confirm){
            dispatch(setError('password y confirm no son identicos'))
            Swal.fire('Error','password y confirm no son identicos','error')
            
            return false
        }
        else if (password.length <= 5){
            dispatch(setError('Password minimo 6 caracter '))
            Swal.fire('Error','Password minimo 6 caracter','error')
            
            return false}
        else{
            dispatch(removeError())
        }
        

        return true

    }

    return (
        <>

            <h3 className="auth__title">Register mock</h3>
            <form onSubmit={handlenRegister}>
                
                
                
                {msgError &&
                 (<div className="auth__alert-error">
                    {msgError}
                </div>)
                } 
              

                <input 
                type="text"
                placeholder='name'
                name="name"
                className="auth__input"
                autoComplete = 'off'
                value = {name}
                onChange={handleInputChange}
                />
                <input 
                type="text"
                placeholder='mail'
                name="email"
                className="auth__input"
                autoComplete = 'off'
                value = {email}
                onChange={handleInputChange}
                />
                <input 
                type="password"
                placeholder='password'
                name="password"
                className="auth__input"
                autoComplete = 'off'
                value = {password}
                onChange={handleInputChange}
                />
                <input 
                type="password"
                placeholder='coconfirm password'
                name="confirm"
                className="auth__input"
                autoComplete = 'off'
                value = {confirm}
                onChange={handleInputChange}
                />


                <button
                type="submit"
                onClick= {handlenRegister}
                className="btn btn-primary btn-block mb-5"
                
                >Register
                
                </button>

    

                <Link 
                to="/auth/login"
                className="link "
                >
                    Â¿Ya estas registrado?
                </Link>


            </form>
            
        </>
    )
}
