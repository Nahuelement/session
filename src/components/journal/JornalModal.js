
import Modal from 'react-modal';


import { useDispatch, useSelector } from 'react-redux';
import { endModal } from '../../actions/modal';




const customStyles = {
    content: {
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
    },
  };


Modal.setAppElement('#root');







export const JornalModal = () => {

    const dispatch = useDispatch()
    
     const {active} = useSelector((state) =>state.modal)

   

   
    

    

    const closeModal = () => {
        dispatch(endModal())
        

    }
    
                
    
    



    return (
        <>
        <Modal
        isOpen={active}
        // onAfterOpen={afterOpenModal}
        onRequestClose={closeModal}
        closeTimeoutMS ={2000}
        style={customStyles}
        className = 'modal'
        overlayClassName = 'modal-fondo'
        >
        <svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="33vw" height="33vh" viewBox="0 0 1024 1024">
        <title></title>
        <g id="icomoon-ignore">
        </g>
        <path d="M0 175.957c0-8.704 0.896-17.109 2.56-25.387l83.456 73.984 91.477-101.547-81.664-72.661c11.051-3.072 22.613-4.864 34.645-4.864 71.707 0.094 129.871 57.944 130.474 129.521v0.953c0 6.485-0.512 12.971-1.408 19.2l101.76 90.112c-36.639 28.444-65.786 64.795-85.151 106.601l-0.737 1.772-105.515-93.397c-11.776 3.909-25.333 6.171-39.416 6.187h-0.008c-72.040-0.049-130.426-58.435-130.475-130.47v-0.005zM893.525 717.568c-13.056 0-25.643 1.92-37.547 5.504l-98.475-87.211c-17.065 45.673-43.592 84.372-77.385 115.253l-0.226 0.203 85.035 75.307c-1.159 6.419-1.84 13.826-1.877 21.385v0.033c0.079 72.047 58.502 130.422 130.56 130.422 13.312 0 26.159-1.992 38.258-5.695l-0.925 0.244-84.992-75.52 91.307-100.907 84.523 75.093c1.451-7.723 2.219-15.531 2.219-23.637v-1.109c-0.604-71.575-58.767-129.365-130.428-129.365-0.017 0-0.033 0-0.050 0h0.003zM748.587 393.643l105.515-93.397c12.459 3.968 25.643 6.187 39.424 6.187 72.029-0.102 130.381-58.516 130.381-130.56 0-8.937-0.898-17.665-2.609-26.097l0.142 0.838-83.456 73.984-91.477-101.547 81.664-72.661c-10.369-3.033-22.289-4.805-34.613-4.864h-0.032c-71.707 0.094-129.871 57.944-130.474 129.521v0.953c0 6.485 0.512 12.971 1.408 19.2l-101.76 90.112c36.634 28.433 65.78 64.769 85.151 106.56l0.737 1.771zM266.453 635.861l-98.432 87.211c-11.25-3.512-24.186-5.535-37.593-5.535-71.672 0-129.841 57.808-130.428 129.341v1.165c0 8.107 0.768 15.915 2.219 23.637l84.565-75.093 91.307 100.907-85.035 75.52c11.175 3.459 24.021 5.452 37.333 5.452 72.058 0 130.482-58.375 130.56-130.415v-0.007c-0.037-7.593-0.718-15-1.992-22.203l0.115 0.784 85.035-75.307c-34.035-31.081-60.576-69.78-76.988-113.438l-0.665-2.018zM649.216 749.227c4.752 7.115 7.584 15.864 7.584 25.275 0 25.355-20.554 45.909-45.909 45.909s-45.909-20.554-45.909-45.909c0-2.536 0.206-5.025 0.601-7.449l-0.036 0.265h-8.192c0.339 2.097 0.533 4.515 0.533 6.978 0 25.355-20.554 45.909-45.909 45.909s-45.909-20.554-45.909-45.909c0-2.463 0.194-4.88 0.567-7.238l-0.034 0.261h-6.4c0.246 1.807 0.386 3.896 0.386 6.018 0 25.889-20.867 46.904-46.699 47.145h-0.023c-25.771 0-46.677-21.333-46.677-47.573 0-8.96 2.432-17.28 6.656-24.405-63.701-48.725-105.557-130.176-105.557-222.421 0-149.205 109.184-270.165 243.755-270.165s243.755 120.96 243.755 270.165c0 92.715-42.283 174.549-106.581 223.147zM470.272 496.896c-5.988-11.237-14.005-20.631-23.646-28.010l-0.204-0.15c-11.16-6.916-24.659-11.083-39.118-11.264h-0.050c-4.829 0.007-9.524 0.567-14.029 1.618l0.418-0.082c-17.28 2.133-31.872 13.952-41.941 27.605-8.832 10.965-15.701 24.277-15.872 38.613-1.365 11.861 1.024 23.723 3.2 35.328 2.816 13.739 12.501 24.917 23.467 33.024 5.077 4.053 12.373 9.643 18.773 4.949 4.267-5.717 0.683-14.080 4.565-20.395 1.536-6.112 5.552-11.078 10.89-13.853l0.118-0.056c11.221-5.632 22.485-11.179 34.475-15.147 9.728-2.859 17.749-9.344 26.027-14.891 10.88-8.405 18.091-23.808 12.928-37.291zM512.768 596.096c-11.904-0.299-26.88 52.779-24.491 75.947 7.467 30.72 10.112-21.547 24.491-21.589 13.781 0.597 11.733 53.547 22.613 21.504 3.328-21.333-9.557-75.52-22.613-75.861zM684.971 560.555c2.176-11.605 4.565-23.467 3.2-35.328-0.171-14.336-7.040-27.648-15.872-38.613-10.069-13.653-24.661-25.429-41.941-27.605-4.087-0.969-8.782-1.529-13.606-1.536h-0.005c-13.653-0.043-27.477 4.267-39.168 11.264-10.027 7.296-17.92 17.323-23.851 28.16-5.163 13.483 2.048 28.885 12.928 37.291 8.277 5.547 16.299 12.032 26.027 14.933 11.947 3.925 23.253 9.472 34.475 15.104 5.456 2.832 9.472 7.797 10.977 13.762l0.031 0.147c3.883 6.272 0.299 14.677 4.523 20.395 6.443 4.693 13.739-0.896 18.773-4.949 11.008-8.107 20.693-19.285 23.509-33.024z"></path>
</svg>

        </Modal>
        
        </>
    )
}
