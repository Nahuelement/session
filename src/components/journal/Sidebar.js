import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startLogout } from '../../actions/auth'
import { openModal } from '../../actions/modal'
import {  startNewNotes } from '../../actions/notes'
import { JournalEntries } from './JournalEntries'



export const Sidebar = () => {


    const dispatch = useDispatch()
    const {name} =  useSelector(state =>state.auth)


    const handleNewEntry = () => {
        dispatch(startNewNotes())
    }
    const handleOpenModal = () => {
        dispatch(openModal())
    }
    const handlendLogout = () => {
        dispatch(startLogout())
        
    }


    return (
       <aside className = 'journal__sidebar'>
           <div className = 'journal__sidebar-navbar'>
            <div style = {{display:'inline'}}>
               <h3 className = 'mt-5'  >
               <i className="fas fa-battery-full fa-1x " style ={{padding:'18px'}}></i>
               <label style = {{display:'inline-block',fontSize: '2.1vh'}}>{name}</label>
               </h3>
            </div>
               <button 
               className = 'btn'
               name = 'logout'
               style ={{marginTop:'-10px'}}
               onClick = {handlendLogout}
               >
                    Salir
               </button>

           </div>

          

           <JournalEntries />
           <div 
           className='journal__new-mock'
           onClick={handleNewEntry}
           >
               <i className="fas fa-users fa-2x"></i>
               
               <p className = 'mt-3'>
                   Nuevo usuario mock
               </p>
           </div>
           <div 
           className='journal__new-entry'
           onClick={handleOpenModal}
           >
               <i className="far fa-check-circle fa-5x"></i>
               
               <p className = 'mt-3'>
                   SOPORTE
               </p>
           </div>

       </aside>
    )
}
