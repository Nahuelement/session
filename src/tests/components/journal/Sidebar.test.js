import { Sidebar } from "../../../components/journal/Sidebar";
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store' 
import thunk from "redux-thunk"
import {mount} from 'enzyme'

import { types } from "../../../types/types";

import { startLogout } from "../../../actions/auth";
import { startNewNotes } from "../../../actions/notes";


jest.mock('../../../actions/notes',() => ({ //mock devuelve un objeto ()
    startNewNotes:jest.fn(), 
   
}))

jest.mock('../../../actions/auth',() => ({ //mock devuelve un objeto ()
    startLogout:jest.fn(), 
   
}))
  

const middlewares = [thunk]
const mockStore = configureStore(middlewares) 


const initialState = { // inicial estado de mock que remplaza al store de redux
    auth:{
        name:'Nahuel'
    },
    
    ui:{
        loading:false,
        msgError:null  
    },
    notes:{
        active:{
            id:'ade98'
        },
        notes:[]
    }
}
let store =mockStore(initialState)//STORE NO UTILIZA LOS REDUCER
store.dispatch = jest.fn()


const wrapper = mount(
    
    <Provider store = {store}>
         <Sidebar />
    </Provider>
)

describe('Pruebas con < Sidebar />', () => {

    test('debe de mostrarse correctamente  ', () => {

        expect(wrapper).toMatchSnapshot()
        
    })
    test('llamar al startLogout ', () => {

        wrapper.find('button[name="logout"]').prop('onClick')() 
        expect(startLogout).toBeCalled()
    })
    test('llamar al startNewNotes ', () => {
        
        const newEntryExpect = wrapper.find('.journal__new-entry')
        newEntryExpect.prop('onClick')()
        expect(startNewNotes).toBeCalled() 
    })
    
    
    
    
})
