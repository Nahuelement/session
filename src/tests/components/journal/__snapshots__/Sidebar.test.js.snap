// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con < Sidebar /> debe de mostrarse correctamente   1`] = `
<aside
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-navbar"
  >
    <div
      style={
        Object {
          "display": "inline",
        }
      }
    >
      <h3
        className="mt-5"
      >
        <i
          className="far fa-address-card fa-2x "
          style={
            Object {
              "padding": "18px",
            }
          }
        />
        <span
          style={
            Object {
              "display": "inline-block",
              "fontSize": "19px",
            }
          }
        >
          Nahuel
        </span>
      </h3>
    </div>
    <button
      className="btn"
      name="logout"
      onClick={[Function]}
      style={
        Object {
          "marginTop": "-10px",
        }
      }
    >
      Salir
    </button>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5"
    >
      Nueva entrada
    </p>
  </div>
  <div
    className="journal__entries animate__animated  animate__fadeInUpBig animate__slow"
  />
</aside>
`;
